apis/v1alpha1/route_types.go

```diff
@@ -35,6 +35,23 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
+const (
+	// AnnotationAppProtocol defines the protocol a Gateway should use for
+	// communication with a Kubernetes Service. The annotation serves as a stop-gap
+	// solution for the Kubernetes versions < 1.18, where the `AppProtocol` field
+	// on the `Service` resource is not available. Using this annotation is highly
+	// discouraged when AppProtocol field is available. AppProtocol field, when
+	// populated, takes precedence over this annotation.
+	//
+	// The annotation value must be a comma separated with each value containing
+	// a port number followed by a forward slash and the protocol.
+	// Examples:
+	//
+	// - `networking.x-k8s.io/app-protocol: 8000/http,8443/https`
+	// - `networking.x-k8s.io/app-protocol: 4000/tcp,5000/tls`
+	AnnotationAppProtocol = "networking.x-k8s.io/app-protocol"
```

Comment:
One limitation here is that if one defines the same port for TCP, UDP, SCTP, one cannot define different protocols for them.
In my experience, this is not common enough but I could be wrong here.

Reply:
To clarify, are you referring to a Service that is exposing all 3 protocols on the same port number?

Reply:
Correct. This is a problem if there is more than one protocol exposed on the same port number.